File: C/342_isPowerOfFour.c
// check if "1" in odd position
bool isPowerOfFour(int num) {
    return (num > 0) && ((num & (num - 1)) == 0) && ((num & 0x55555555) == num); 
}

bool isPowerOfFour_2(int num) {
    return ((num - 1) & num) == 0 && (num - 1) % 3 == 0; 
}

File: python/342_isPowerOfFour.py
class Solution(object):
    def isPowerOfFour(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num <= 0: return False
        if num == 1: return True
        
        one = 0
        for b in bin(num - 1)[2:]:
            if b is '0':
                return False
            one += 1
        
        return one % 2 == 0
        
